portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,20 +13,55 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
+import com.google.gson.Gson;
 
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import java.util.ArrayList;
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private ArrayList<Comment> comments = new ArrayList<>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Juan!</h1>");
+    response.setContentType("application/json;");
+    String json = getCommentsJson();
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Comment newComment = getComment(request);
+    comments.add(newComment);
+    response.sendRedirect("/index.html");
+  }
+
+  public Comment getComment(HttpServletRequest request) {
+    String firstName = request.getParameter("inputFirstName");
+    String lastName = request.getParameter("inputLastName");
+    String message = request.getParameter("inputComment");
+    return new Comment(firstName,lastName,message);
+  }
+  
+  public String getCommentsJson() {
+      return (new Gson()).toJson(comments);
   }
 }
+
+class Comment {
```

Comment:
Personally, I like classes to be defined before they are referenced, so for example in this case I would put `Comment` above the `Servlet` definition. It's probably because I used to write a lot of C++ and it complains if you don't do that. But! I know people that write a lot of Java and they prefer these kinds of data class definitions to come later. What do you think is easier to follow? I can see arguments for both sides.

Reply:
Typically when I write classes that are small, like the Comment class, I will just leave them at the bottom since it's only a few lines of code that assigns variables. I do see the argument for leaving it at the top so whoever is reading the code will know the structure of a comment before seeing any code that uses comment, but in this case since the comment class is so self-explanatory I don't think it matters much.